### ISSUE TITLE
Visual Scripting suggestions

### ISSUE THREAD

**Issue by cbscribe** [godotengine/godot-docs#466]
This is to discuss @reduz 's new VisualScripting intro.  While thorough, it is very long and covers a *lot* of info for its intended audience(s).

As I see it there are 3 separate concepts being covered:
- What is VisualScripting? (the intro part that defines and defends its existence) This is all a bit abstract and not relevant for beginners - they don't understand the issues.
- Getting started - creating your first VS, a walkthrough of how to create a simple script
- List of various nodes & their purposes. I feel like this terminology might get confusing - "Node" already has a very different meaning in Godot

This suggest to me splitting it into separate docs along these lines.  I'd love to hear everyone's thoughts, since we're also reworking the GDScript intros right now as well - it's a good opportunity to address some of the same issues and perhaps tie them more closely together.

**Comment by NathanLovato** [comment]
Definitely agree. @reduz did the writing work, now it's up to us to edit it. I don't think he'll mind if we chop it up into shorter pages (the original is close to 4k words and 50 pics, right?). We can still start with some edit/proofing pass before we create 3 articles out of it. At least, I have too much on my plate to do it all now

**Comment by reduz** [comment]
Keep in mind this is mostly a reference.. step by step will include visual
script exames

**Comment by NathanLovato** [comment]
It it both a step by step tutorial and a reference at the moment. It's worth splitting into 2 pages. Just so that someone looking for the reference can open the ref page and use the TOC to access nodes quickly. 

Also yeah, considering Nodes are so important in Godot and in the docs, I'd vouch for a different name, e.g. block. It may definitely confuse people otherwise.

For the step-by-step example, how about working with chris's demo project? He wrote a new "your first game" tutorial already for the docs, that shows some of Godot's power and core concepts. It's a simple yet complete game:

- thus a good fit for VS
- a good opportunity for new users to compare GDscript and VS
- the assets are ready and there's the gameplay available for reference

**Comment by reduz** [comment]
All the API uses Node terminology so I guess it's difficult to change.. I
am not sure how much of a misunderstanding it can be.

About turorials, I really want to see how possible it is to make a single
tutorial bit show code snippets in all GDScript, VisualScript, C# and C++..
we were checking this and found it is possible with the current system, but
did not check it yet

**Comment by reduz** [comment]
ok, so it's perfectly possible to do it it seems, here is a test:
http://docs.godotengine.org/en/latest/learning/step_by_step/scripting.html#handling-a-signal

The question is them should we just make a single tutorial that supports
all scriptings, or should each use individual tutorials? IMO the workflow
is pretty much the same with all languages so I would just think the best
is to simply use a single tutorial..

**Comment by NathanLovato** [comment]
Single tutorial +1
For the "nodes" then, let's just add a short info note that in the context of VS, we're talking of _graph nodes_, should be okay

**Comment by cbscribe** [comment]
The tab method for showing snippets is great for reference docs, but I don't think it works for step-by-step tutorials.

I've implemented the beginner game in VS here: https://github.com/kidscancode/Godot3_Dodge_VS

I don't think it's at all feasible to have one tutorial that covers both GDScript and VS. The written instructions will be very different. As you know, @NathanLovato writing step-by-step instructions is not easy, and shouldn't confuse the reader with more than one way to do things. For example, to avoid a giant mess of spaghetti, the VS version needs a lot more separation of code into individual functions. It actually becomes a very different process than building the game in GDScript.

I'm going to add an issue to write up the starter game in VS, and see if we can come up with a way to link them. For example, the node setup is identical, it's only once you start scripting that things get really different.

**Comment by NathanLovato** [comment]
Makes total sense. Feel free to do it as a separate page and cross-reference between the two versions

**Comment by cbscribe** [comment]
Closing as there's been no activity for some time. Feel free to open a new issue if you feel the need is still there.
